app/models/mixins/archived_mixin.rb

```diff
@@ -4,6 +4,11 @@ module ArchivedMixin
   included do
     scope :archived, -> { where.not(:deleted_on => nil) }
     scope :active, -> { where(:deleted_on => nil) }
+
+    # records are active from the date where they were created until the date they are deleted
+    scope :active_for, (lambda do |timestamp|
```

Comment:
This name is hard to wrap my head around.  Probably, because active doesn't make sense to me for anything that has been deleted.

How about the defining the opposite:
`deleted on older than timestamp`. Maybe call it `archived_older_than` and just use a `.not`?

Reply:
Naming is hard.

Reply:
or `archived_before` and use `.not.archived_before(timestamp)`

Reply:
I added the first scope. not sure how to get a scope and a not working together.
(It almost feels like we could use `not.active` for the `archived`.)

I can't figure out how to use a `not` with a scope. All of my attempts are invalid. I haven't even tried to do this with a parameter.

```ruby
Container.where.not.active
Container.where.not(Container.active)
```

Reply:
I don't like this new naming - the new name is complicated to me and the old one wasn't.  Naming is hard™️  😆 

Let's discuss over voice - @jrafanie I'm having trouble understanding what was confusing about the original wording.

Reply:
> Let's discuss over voice - @jrafanie I'm having trouble understanding what was confusing about the original wording.

`active_for` including recently deleted entries along with not deleted is strange.  Maybe `recently_active_since` would be fine if you prefer this logic.